File: app/src/main/java/com/example/shoppingapp/adapters/CategoryAdapter.java
package com.example.shoppingapp.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.util.Pair;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shoppingapp.fragments.ItemsScreen;
import com.example.shoppingapp.util.Animation;
import com.example.shoppingapp.Category;
import com.example.shoppingapp.MainActivity2;
import com.example.shoppingapp.R;

import java.util.List;


public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder> {


    List<Category> categoryList;
    Context context;

    public CategoryAdapter(List<Category> categoryList, Context context) {
        this.categoryList = categoryList;
        this.context = context;
    }
    public CategoryAdapter(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View v=layoutInflater.inflate(R.layout.category_layout,parent,false);
        return new CategoryViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        holder.category_layout.setOnClickListener(v ->  {


            Pair<View, String> p1 = Pair.create(holder.categoryImg,  context.getString(R.string.CATEGORY_IMG_TRANSITION_TAG));
            Pair<View, String> p2 = Pair.create(holder.categoryName,  context.getString(R.string.CATEGORY_NAME_TRANSITION_TAG));
            Pair<View, String> p3 = Pair.create(holder.category_layout,  context.getString(R.string.CATEGORY_Layout_TRANSITION_TAG));

            Intent intent=new Intent(context, MainActivity2.class);
            intent.putExtra(context.getString(R.string.FRAGMENT_NAME_TAG), ItemsScreen.class.getSimpleName());

            Animation.MultipleSharedElementTransition(
                    context,
                    intent,
                    p3,p2,p1);
        });

        if(categoryList==null) return;
//        holder.categoryImg.setImageAlpha(categoryList.get(position).getIcon());
        holder.categoryName.setText(categoryList.get(position).getName());

    }

    @Override
    public int getItemCount() {
        return 10;
    }

    static class CategoryViewHolder extends RecyclerView.ViewHolder{


        ImageView categoryImg;

        TextView categoryName;

        LinearLayout category_layout;

        public CategoryViewHolder(@NonNull View itemView) {
            super(itemView);

            categoryImg = itemView.findViewById(R.id.category_img);
            categoryName = itemView.findViewById(R.id.category_txt);
            category_layout = itemView.findViewById(R.id.category_layout);
        }
    }


}

File: app/src/customer/java/com/example/shoppingapp_customer/adapters/CategoryAdapter.kt
package com.example.shoppingapp_customer.adapters

import android.content.Context
import android.content.res.TypedArray
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.core.util.Pair
import androidx.navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import com.example.shoppingapp.R
import com.example.shoppingapp_customer.adapters.CategoryAdapter.CategoryViewHolder
import com.google.android.material.card.MaterialCardView


class CategoryAdapter(val context: Context) : RecyclerView.Adapter<CategoryViewHolder>() {

    private var categoryList: Pair<Array<String>, TypedArray> = Pair(context.resources.getStringArray(R.array.category_name),
            context.resources.obtainTypedArray(R.array.category_icon))

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val inflater = LayoutInflater.from(this.context)
                .inflate(R.layout.category_card_layout, parent, false)
        return CategoryViewHolder(inflater)
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        holder.category_layout.setOnClickListener {


            val bundle= Bundle()
            bundle.putInt(context.getString(R.string.SELECTED_CATEGORY_TAG),position)
            it.findNavController().navigate(R.id.action_homeScreen_to_itemsScreen,bundle)



        }

        holder.categoryImg.setImageResource(categoryList.second?.getResourceId(position,R.drawable.error)!!)
        holder.categoryName.text = categoryList.first?.get(position)
    }

    override fun getItemCount(): Int {
        return categoryList.first?.size!!
    }

    class RoundCategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var categoryImg: ImageView
        var categoryName: TextView
        var category_layout: LinearLayout

        init {
            categoryImg = itemView.findViewById(R.id.category_img)
            categoryName = itemView.findViewById(R.id.category_txt)
            category_layout = itemView.findViewById(R.id.category_layout)
        }
    }

    class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var categoryImg: ImageView
        var categoryName: TextView
        var category_layout: MaterialCardView

        init {
            categoryImg = itemView.findViewById(R.id.img_category_ccl)
            categoryName = itemView.findViewById(R.id.txt_category_ccl)
            category_layout = itemView.findViewById(R.id.card_category_ccl)
        }
    }

}